src/helpers/images.js

```diff
@@ -0,0 +1,95 @@
+// eslint-disable-next-line import/no-unresolved
+const { createRemoteFileNode } = require('gatsby-source-filesystem')
+const { logInfo } = require('./logger')
+
+const supportedExtensions = ['jpeg', 'jpg', 'png', 'webp', 'tif', 'tiff']
+
+/**
+ * Recursively search for image fields in content
+ *
+ * @param {any} entry
+ * @return {array} an Array of all node with { contentType: 'image/png', url: ... }
+ */
+const locateImageFields = entry => {
+  if (typeof entry !== 'object' || entry == null) return []
+
+  let results = []
+
+  // perfect condition
+  const { contentType, url } = entry
+  if (contentType && url) {
+    const ext = contentType.split('/')[1]
```

Comment:
@d4rekanguok What is your thoughts around check for the specific `supportedExtensions` as the last part of the content type versus checking the first part, ie. checking for any content type that starts with `image/`?

Reply:
At the time I thought it made sense to only download image type that `sharp` supports. But now come to think about it, perhaps it's better to download any `media` or `image` files locally, and then let `gatsby-transformer-sharp` decide whether to transform the file or not. We won't need to maintain a list of support extensions then!

Reply:
Yes, I think that is a good idea. Then if the `gatsby-transformer-sharp` plugin rolls out more support, we get it by default.

Reply:
I've quickly modified to fix this